packages/dx-react-grid/src/components/table-layout.jsx

```diff
@@ -91,13 +91,20 @@ export class TableLayout extends React.PureComponent {
       const { columns } = this.props;
       const { sourceColumnIndex, targetColumnIndex } = this.state;
 
-      if (sourceColumnIndex === -1 || targetColumnIndex === -1) return columns;
+      let result = columns;
 
-      const result = columns.slice();
+      const staticWidth = columns.filter(column => column.width === undefined).length === 0;
```

Comment:
May be it is better to call this const `isStaticWith` or `isFixedWidth` so its name could reflect its value type?